### PR TITLE
Added CurrencyFieldInput design

### LINKED ISSUE #4123: New CurrencyFieldInput design
# Desired Behavior

image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=3007-69739&mode=design&t=qpJNVAebuHYFQT5A-11

Clicking on "USD" should open a dropdown menu with a search. The default value should be set by default:

image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=23581-64924&mode=design&t=qpJNVAebuHYFQT5A-11

### PR DESCRIPTION
**Pull request by jeet1desai**
Closes #4123 

[#4254.webm](https://github.com/twentyhq/twenty/assets/52026385/23033d1b-3c26-4762-aa80-360ca73577e7)

### CODE REVIEW

**Comment by magrinj** [comment]
Hello @jeet1desai, thanks for your contribution, is it still a work in progress as the PR is marked as a draft ?

**Comment by jeet1desai** [comment]
Hello @magrinj, It's still not completed. I need support if you can help me.

Whenever I try to edit and select a different currency code, it defaults to USD. I have also attached the video.

[defaults to USD.webm](https://github.com/twentyhq/twenty/assets/52026385/0899ecdb-c8ce-476c-92dc-e91649766ca2)

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/CurrencyFieldInput.tsx [hunk 1]
```diff
@@ -79,19 +79,28 @@ export const CurrencyFieldInput = ({
     });
   };
 
+  const handleSelect = (newValue: string) => {
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/CurrencyFieldInput.tsx:82]: All newValue args should be of type `Currency` now.

```ts
const handleEnter = (newCurrencyValue: Currency) => {
```

**Review by lucasbordeau — COMMENTED** [review]
Please see in `computeDraftValueFromFieldValue.ts` 

```ts 
return {
  amount: fieldValue?.amountMicros ? fieldValue.amountMicros / 1000000 : '',
  currenyCode: CurrencyCode.USD, // <== Maybe change this hard coded value to something dynamic here ? Example : fieldValue.currencyCode ?
} as unknown as FieldInputDraftValue<FieldValue>;
```

**Comment by jeet1desai** [comment]
> Please see in `computeDraftValueFromFieldValue.ts`
> 
> ```ts
> return {
>   amount: fieldValue?.amountMicros ? fieldValue.amountMicros / 1000000 : '',
>   currenyCode: CurrencyCode.USD, // <== Maybe change this hard coded value to something dynamic here ? Example : fieldValue.currencyCode ?
> } as unknown as FieldInputDraftValue<FieldValue>;
> ```

Thank you @lucasbordeau, It worked

**Review by lucasbordeau — CHANGES_REQUESTED** [review]
@jeet1desai Nice work.

Some minor differences : 

Figma : 

image

React app : 

image

- Currency icon is too thick
- Currency text is too light, the font-weight is 500 in Figma
- The border between the currency text and the input should be a clean vertical bar

**Review by lucasbordeau — APPROVED** [review]

**Pull request merged**